### ISSUE TITLE
Checked parent - unchecked child doesn't behave correctly for decoupleChildFromParent = true

### ISSUE THREAD

**Issue by tschmidt01** [leovo2708/ngx-treeview#85]
data structure:
`itCategory = [new TreeviewItem({
    text: 'IT', value: 9, checked: true, children: [
      {
        text: 'Programming', value: 91, children: [], checked: false
      }
    ]
  })];`

config:
`    this.ngxTreeviewConfig = {
      hasAllCheckBox: false,
      hasFilter: false,
      hasCollapseExpand: true,
      decoupleChildFromParent: true,
      maxHeight: 600
    };`

view:
`<ngx-treeview [config]="ngxTreeviewConfig" [items]="itCategory" (selectedChange)="onSelectedChange($event)"></ngx-treeview>`

results in:
![screen shot 2017-12-13 at 17 28 12](https://user-images.githubusercontent.com/15064827/33949829-0658d52c-e02b-11e7-978e-0169ded5c454.png)

**Comment by jenkyjeni** [comment]
I am also facing same issue. any Solution around ?

**Comment by leovo2708** [comment]
Yes, when using decoupleChildFromParent =true, any changes from parent will not reflect to children.
Please change decoupleChildFromParent =false

**Comment by jenkyjeni** [comment]
In my case, i need to check the parent by default when there is not child selected

**Comment by cjosue15** [comment]
I have the same issue, any response?

**Comment by leovo2708** [comment]
I am upgrading to Angular 10. When completed, I will verify your issue.

**Comment by cjosue15** [comment]
Thanks a lot, ngx-trevieew is very useful

**Comment by leovo2708** [comment]
> itCategory = [new TreeviewItem({
>     text: 'IT', value: 9, checked: true, children: [
>       {
>         text: 'Programming', value: 91, children: [], checked: false
>       }
>     ]
>   })];

Why we need parent node is checked while child nodes are unchecked? ngx-treeview will correct states of nodes before rendering. So, you will see the parent node is unchecked although you set it is checked.

**Comment by silambarasan-r** [comment]
@leovo2708 
as you're asking, so what is the purpose of `decoupleChildFromParent` ?

Use cases may change as per the user needs. 
It'd be great if you give the option to change that too. Thanks in advance.
